core/src/databases/database.rs

```diff
@@ -16,9 +40,86 @@ impl Database {
             data_source_id: data_source_id.to_string(),
             database_id: database_id.to_string(),
             name: name.to_string(),
+            db_type: DatabaseType::LOCAL,
+        }
+    }
+
+    pub async fn get_schema(
+        &self,
+        project_id: i64,
+        store: Box<dyn Store + Sync + Send>,
+    ) -> Result<DatabaseSchema> {
+        match self.db_type {
+            DatabaseType::REMOTE => Err(anyhow!("Remote DB not implemented.")),
+            DatabaseType::LOCAL => {
+                let project = Project::new_from_id(project_id);
+                let (tables, _) = store
+                    .list_databases_tables(&project, &self.data_source_id, &self.database_id, None)
+                    .await?;
+                let (rows, _) = store
+                    .list_database_rows(
+                        &project,
+                        &self.data_source_id,
+                        &self.database_id,
+                        &None, // get rows from all tables
```

Comment:
I don't think we want to allow this. This seems a bit off, not a very clean interface. We can always loop in parallel on tables and get process them independently. You'll never want to do something where you actually need rows coming from different tables.

Reply:
Well in this specific case, you can get everything in one query VS having to do one per table. What is the upside of doing N queries in parallel here ?

Reply:
Better interfaces :)

Reply:
Simpler postgres.rs code also

Reply:
Fixed !

Reply:
went with a `tokio::join`